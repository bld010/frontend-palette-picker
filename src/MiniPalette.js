import React from 'react';
import './MiniPalette.scss';

const MiniPalette = (props) => {
  let { colors } = props.palette;


  return(
    <div onClick={() => props.setCurrentPalette(props.palette)} className="MiniPalette">
      <div style={{backgroundColor: colors[0].hex}}></div>
      <div style={{backgroundColor: colors[1].hex}}></div>
      <div style={{backgroundColor: colors[2].hex}}></div>
      <div style={{backgroundColor: colors[3].hex}}></div>
      <div style={{backgroundColor: colors[4].hex}}></div>
    </div>
  )
}

export default MiniPalette;