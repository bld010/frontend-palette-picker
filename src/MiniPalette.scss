.MiniPalette {
  width: 90%;
  height: 90px;
  margin-top: 10px;
  display: flex;
  flex-direction: row;

  & div {
    height: 80px;
    width: 80px;
    border-radius: 50%;
    margin-right: -20px;
  }
}