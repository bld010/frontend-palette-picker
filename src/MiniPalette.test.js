import React from "react";
import { shallow } from "enzyme";
import MiniPalette from "./MiniPalette";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("Mini Palettes", () => {

  let mockOnKeyDown = jest.fn();
  let mockSetCurrentPalette = jest.fn();
  let mockPalette = {
    id: 125,
    colors: [{hex: "#00ed3f"}, {hex: "#00ed3f"},{hex: "#00ed3f"},{hex: "#00ed3f"},{hex: "#00ed3f"}],
    name: "1",
    folder_id: 84,
    created_at: "2019-10-15T01:01:16.495Z",
    updated_at: "2019-10-15T01:01:16.495Z"
  };

  let wrapper = shallow(<MiniPalette onKeyDown={mockOnKeyDown} setCurrentPalette={mockSetCurrentPalette} palette={mockPalette} />)

  it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
  })

  it('should fire setCurrentPalette when clicked', () => {
      wrapper.find('.MiniPalette').simulate('click')
      expect(mockSetCurrentPalette).toHaveBeenCalled()
  })
});
