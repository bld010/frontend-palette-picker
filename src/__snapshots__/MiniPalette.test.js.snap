// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mini Palettes should match the snapshot 1`] = `
<div
  className="MiniPalette"
  onClick={[Function]}
>
  <div
    style={
      Object {
        "backgroundColor": "#00ed3f",
      }
    }
  />
  <div
    style={
      Object {
        "backgroundColor": "#00ed3f",
      }
    }
  />
  <div
    style={
      Object {
        "backgroundColor": "#00ed3f",
      }
    }
  />
  <div
    style={
      Object {
        "backgroundColor": "#00ed3f",
      }
    }
  />
  <div
    style={
      Object {
        "backgroundColor": "#00ed3f",
      }
    }
  />
</div>
`;
